harvester_e2e_tests/scenarios/test_rancher_integration.py

```diff
@@ -130,9 +133,19 @@ def _wait_for_cluster_ready_status():
             _wait_for_cluster_ready_status,
             step=5,
             timeout=request.config.getoption('--rancher-cluster-wait-timeout'))
-    except polling2.TimeoutException as e:
-        errmsg = 'Timed out while waiting to import Harvester.'
-        raise AssertionError(errmsg) from e
+    except polling2.TimeoutException:
+        from datetime import datetime
+        from json import dumps
+        fmt = "%Y-%m-%dT%H:%M:%SZ"
+
+        def _key(item):
+            t = item['lastUpdateTime']
+            return datetime.strptime(t, fmt) if t else datetime(2000, 1, 1)
+
+        items = sorted(cluster_json['status']['conditions'], key=_key)
```

Comment:
We don't necessarily need to change this :smile: - this is more of a general comment:

There is the chance within the `cluster_json` dict we may also run into the possibility of it not having either 'status' or 'conditions'.  If that happened it would throw a `KeyError` within the assignment of `items` - we could possibly specify defaults like I mentioned on the other comment below.

https://docs.python.org/3/library/stdtypes.html#dict.get

Reply:
The key will never missed, or it would be a critical issue for the cluster.
If you have more concern, you can test it on the API endpoint `https://<your_rancher_URL>/v1/provisioning.cattle.io.clusters/fleet-default/<The_cluster_ID>`.

By the case, it will apply to RKE1 and RKE2 cluster which we attempt to create.